Adds CI/CD overview page (#44137)
openshift/openshift-docs: cicd/index.adoc

:_content-type: ASSEMBLY
[id="ci-cd-overview"]
= {product-title} CI/CD overview
include::_attributes/common-attributes.adoc[]
:context: ci-cd-overview

toc::[]

> {product-title} is an enterprise ready Kubernetes platform for developers, which enables organizations to automate application delivery process through DevOps practices, such as continuous integration (CI) and continuous delivery (CD). To meet your organizational needs, the {product-title} provides the following CI/CD solutions:

Reviewer: ```suggestion
{product-title} is an enterprise-ready Kubernetes platform for developers, which enables organizations to automate the application delivery process through DevOps practices, such as continuous integration (CI) and continuous delivery (CD). To meet your organizational needs, the {product-title} provides the following CI/CD solutions:
```

Author: @rolfedh Fixed.

[id="openshift-builds"]
== OpenShift Builds
With OpenShift Builds, you can create cloud-native apps by using a declarative build process. You can define the build process in a YAML file that you use to create a BuildConfig object. This definition includes attributes such as build triggers, input parameters, and source code. When deployed, the BuildConfig object typically builds a runnable image and pushes it to a container image registry.

OpenShift Builds provides the following extensible support for build strategies:

* Docker build
* Source-to-image (S2I) build
* Custom build

For more information, see xref:../cicd/builds/understanding-image-builds.adoc#understanding-image-builds[Understanding image builds]

[id="openshift-pipelines"]
== OpenShift Pipelines
OpenShift Pipelines provides a Kubernetes-native CI/CD framework to design and run each step of the CI/CD pipeline in its own container. It can scale independently to meet the on-demand pipelines with predictable outcomes.

For more information, see xref:../cicd/pipelines/understanding-openshift-pipelines.adoc#understanding-openshift-pipelines[Understanding OpenShift Pipelines]

[id="openshift-gitops"]
== OpenShift GitOps

> OpenShift GitOps is a Red Hat OpenShift operator that uses Argo CD as the declarative GitOps engine that enables GitOps workflows across multicluster OpenShift and Kubernetes infrastructure. GitOps defines infrastructure and application definitions as code. Then, it uses this code to manage multiple workspaces and clusters to simplify the creation of infrastructure and application configurations. GitOps also handles and automates complex deployments at a fast pace, saving time during deployment and release cycles.

Reviewer: ```suggestion
OpenShift GitOps is an Operator that uses Argo CD as the declarative GitOps engine. It enables GitOps workflows across multicluster OpenShift and Kubernetes infrastructure. GitOps defines infrastructure and application definitions as code. Then, it uses this code to manage multiple workspaces and clusters to simplify the creation of infrastructure and application configurations. GitOps also handles and automates complex deployments at a fast pace, saving time during deployment and release cycles.
```
I recommend further pruning beyond what I've suggested here. 

This paragraph feels problematic, overloaded, and mixes conceptual information with sequential/procedural information. 

The reader only needs a *short* overview here. They will go to the respective sections for more detailed information.

Author: @rolfedh Please take a look and let me know if it makes sense ?

> OpenShift GitOps is an Operator that uses Argo CD as the declarative GitOps engine. It enables GitOps workflows across multicluster OpenShift and Kubernetes infrastructure. Using OpenShift GitOps, administrators can consistently configure and deploy Kubernetes-based infrastructure and applications across clusters and development lifecycles.

For more information, see xref:../cicd/gitops/understanding-openshift-gitops.adoc#understanding-openshift-gitops[Understanding OpenShift GitOps]

[id="jenkins-ci-cd"]
== Jenkins

> Jenkins automates the process of building, testing, and deploying applications and projects. OpenShift Developer Tools provides a Jenkins image that integrates directly with the {product-title}. Jenkins can be deployed on OpenShift using the templates provided by the Samples Operator, or with the certified Helm chart.

Reviewer: ```suggestion
Jenkins automates the process of building, testing, and deploying applications and projects. OpenShift Developer Tools provides a Jenkins image that integrates directly with the {product-title}. Jenkins can be deployed on OpenShift by using the Samples Operator templates or certified Helm chart.
```

Author: @rolfedh Done.
